### PR TITLE
[Linux] Dynamically set expected guest id and full name for checking guest info

### PR DESCRIPTION
**Pull request by keirazhang**
Since more new guest ids were added in newer VMware Tools, which will be unmapped on older ESXi version servers. This fix dynamically looked for an exactly matched guest id or a closest guest id for the guest OS distribution. 

If it doesn't find an exactly matched or closest guest if from ESXi supported guest ids list, then looking for a proper other Linux guest id according to its kernel version.

`match_guest_id.yml` will match an expected guest id based on guest short-name value in tools.conf or OS release info.
`map_guest_id_to_fullname.yml` will retrieve the guest full name in ESXi guest config options by guest id.

However, it still could have unmapped guest ids when a guest can be identified by newer VMware Tools but not recognized on an old ESXi server. In such case, this case will take it as expected. Expected unmapped guest ids are checked in `check_unmapped_guest_id.yml`

### CODE REVIEW

File: linux/open_vm_tools/get_install_uninstall_cmd.yml [hunk 1]
```diff
@@ -29,14 +29,30 @@
...
     - guest_os_family == "RedHat"
     - guest_os_ansible_pkg_mgr == "yum"
+  block:
+    - name: "Set OS commands for installing or uninstalling packages on {{ vm_guest_os_distribution }}"
+      ansible.builtin.set_fact:
+        package_install_cmd: "yum install -y"
+        package_uninstall_cmd: "yum remove -y"
+        check_update_cmd: "yum check-update"
+        clean_cache_cmd: "yum clean all"
+      when: >-
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply linux/open_vm_tools/get_install_uninstall_cmd.yml]: Updated. Thanks!

File: linux/check_os_fullname/map_guest_id_to_fullname.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- name: "Check guest full name is defined for guest id {{ expected_guest_id }}"
+  ansible.builtin.assert:
+    that:
+      - guest_config_options.guest_id is defined
+      - guest_config_options.guest_id == expected_guest_id
+      - guest_config_options.guest_fullname is defined
+      - guest_config_options.guest_fullname
+    fail_msg: >-
+      Failed to find guest full name for guest id {{ expected_guest_id }} on ESXi {{ esxi_version }}
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply linux/check_os_fullname/map_guest_id_to_fullname.yml:20]: Added hardware version info.

File: linux/check_os_fullname/check_unmapped_guest_id.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+# VMware Tools version is 12.0.0 or later, and ESXi version is 7.0.3 or earlier
+- name: "Set fact to ignore unmapped guest id for {{ vm_guest_os_distribution }} on ESXi {{ esxi_version }}"
+  ansible.builtin.set_fact:
+    ignore_unmapped_guest_id: true
+  when:
+    - guest_os_ansible_distribution in ['AlmaLinux', 'Rocky']
+    - vmtools_version is version('12.0.0', '>=')
+    - esxi_version is version('7.0.3', '<=')
+
+# Guest idsof ProLinux, MIRACLE LINUX, Pardus, Kylin Linux or FusionOS are mapped to VM configured guest id when
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply linux/check_os_fullname/check_unmapped_guest_id.yml]: Fixed. Thanks

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**